### PR TITLE
feat(pf4-table): Introduce new package @patternfly/react-table for table implementation

### PR DESCRIPTION
**Pull request by karelhala**
This PR adds new empty package [@patternfly/react-table](https://www.npmjs.com/login?next=/package/@patternfly/react-table) as place where our implementation of different tables will be located. 

The purpose of splitting the PR is to have less code to review and option to include this package in documentation without any errors.

### CODE REVIEW

File: packages/patternfly-4/react-table/package.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+  "name": "@patternfly/react-table",
+  "version": "0.0.1",
```
Comment (jschuler) [inline packages/patternfly-4/react-table/package.json]: I think to make semantic release versioning work it has to start with 1.0.0

**Comment by dmiller9911** [comment]
You mentioned errors in docs unless the package is published first. Did you try it beforehand? Yarn and Lerna should be able to link the package whether it is published or not. It should not need an empty publish first. I did this a while back, never did finish my branch, but it worked:

https://github.com/dmiller9911/patternfly-react/blob/react-syntax-highlighter/packages/react-docs/package.json#L15

**Comment by karelhala** [comment]
@dmiller9911 I was hoping for lerna and yarn to solve this as well, it looks like not. But speaking of it I def forgot to run `lerna bootstrap` that should do the trick. I'll try it and see what it brings. I've also tried `yarn add file:PATH` but without luck as well (well gatsy was complaining).

The PR split is nice to have anyway, am I right? This PR is huge, I was out for a week and it took me couple of minutes to get around and such locally let alone on GH review.

**Comment by jschuler** [comment]
@karelhala You can see the changes Dan had to make in #868 in react-docs to get the charts package tied in

**Comment by karelhala** [comment]
@jschuler yeah, I can see them and I have similiar thing prepared for https://github.com/patternfly/patternfly-react/pull/899, ut I can move them here if that's preferable.

**Comment by dmiller9911** [comment]
Shouldn't hurt anything. We will just have an empty v1 not that is matters honestly.

**Comment by jschuler** [comment]
Just merge conflicts and then let's get it in

**Comment by karelhala** [comment]
ping @jschuler

**Pull request merged**